build.gradle

```diff
@@ -326,6 +326,7 @@ project(":samza-sql_$scalaSuffix") {
 
   dependencies {
     compile project(':samza-api')
+    compile project(":samza-core_$scalaSuffix")
```

Comment:
Not sure if it's OK to introduce samza-core dependency on sql (for split deploys). Even if it is, I'd prefer to not do that for modularity. Can we move the util to API instead?

Reply:
samza-kafka depends on samza-core, and samza-sql depends on samza-kafka (although I'm not sure why), so samza-sql already depended on samza-core transitively. Just making it explicit.
Also, samza-sql was already using `DelegatingSystemDescriptor`, which is from samza-core.
Agreed that it would be nice for samza-sql to not depend on samza-core, but it also seems like this class doesn't belong in samza-api directly. Maybe our future plans to reorg the modules (e.g. introduce a samza-common) can move this over?

Reply:
Makes sense to make dependency explicit, we can clean up later.